# Predict Box Office Success!
## 3 MOVIES

In [72]:
install.packages("tidyverse")
require(tidyverse)

movies_df <- read.table("/Users/jmanderson/statistics/02_intro-to-regression/ARM by Iain - Exercises/data/movies.csv", 
                       sep=",", header = T, stringsAsFactors = F)


The downloaded binary packages are in
	/var/folders/tz/875z4smn2sz0kndfn6lyfvf40000gn/T//Rtmp02DJQq/downloaded_packages


Loading required package: tidyverse

── Attaching packages ─────────────────────────────────────── tidyverse 1.3.0 ──

✔ ggplot2 3.3.0     ✔ purrr   0.3.4
✔ tibble  3.0.1     ✔ dplyr   0.8.5
✔ tidyr   1.0.2     ✔ stringr 1.4.0
✔ readr   1.3.1     ✔ forcats 0.5.0

── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()



In [8]:
head(movies_df)

Movie                                                         Box      Rate
1 Star Wars: Episode IV - A New Hope (1977)                     460.9357 8.8 
2 Toy Story 3 (2010)                                            414.9845 8.5 
3 Spider-Man 2 (2004)                                           373.3779 7.5 
4 Finding Nemo (2003)                                           339.7144 8.1 
5 The Lion King (1994)                                          328.4230 8.3 
6 Pirates of the Caribbean: The Curse of the Black Pearl (2003) 305.3887 8.0 
  User    Meta Len Win Nom
1 423.838 91   121 36  22 
2 201.748 92   103 35  41 
3 191.179 83   127 15  39 
4 263.202 89   100 35  36 
5 233.977 83    89 27  22 
6 355.801 63   143 28  74

In [28]:
lmRateUserMeta <- lm(Box~Rate + User + Meta, movies_df)
summary(lmRateUserMeta) 


Call:
lm(formula = Box ~ Rate + User + Meta, data = movies_df)

Residuals:
    Min      1Q  Median      3Q     Max 
-49.202 -22.749  -3.598   7.266  90.059 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)   
(Intercept) -169.0862    92.0925  -1.836  0.08055 . 
Rate          35.4962    18.9956   1.869  0.07569 . 
User           0.4328     0.1472   2.940  0.00783 **
Meta           1.2462     0.8047   1.549  0.13640   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 39.53 on 21 degrees of freedom
Multiple R-squared:  0.8841,	Adjusted R-squared:  0.8675 
F-statistic: 53.37 on 3 and 21 DF,  p-value: 5.35e-10


In [41]:
regressionEquation <- function(rate, user, meta){
    return (-169.0862+35.4962*rate+0.4328*user+1.2462*meta)
}

In [42]:
regressionEquation(8.8,423.838,91)

[1] 440.1216

In [75]:
filter(movies_df, Rate>8, Rate<9)

Movie                                     Box      Rate User    Meta Len Win
1 Star Wars: Episode IV - A New Hope (1977) 460.9357 8.8  423.838 91   121 36 
2 Toy Story 3 (2010)                        414.9845 8.5  201.748 92   103 35 
3 Finding Nemo (2003)                       339.7144 8.1  263.202 89   100 35 
4 The Lion King (1994)                      328.4230 8.3  233.977 83    89 27 
5 Up (2009)                                 292.9796 8.3  222.884 88    96 56 
6 Jaws (1975)                               260.0000 8.2  190.092 79   124 12 
  Nom
1 22 
2 41 
3 36 
4 22 
5 32 
6 13

We expect Box office recipets to increase by 35.5 Million dollars for each increase in average user rating (out of 10) , holding all other variables constant. This expectation is only held for Box office receipts between 100 and 460 Million dollars and Ratings between 5.3 and 8.8.

In [90]:
summary(movies_df)

    Movie                Box             Rate            User       
 Length:25          Min.   :100.3   Min.   :5.300   Min.   : 25.51  
 Class :character   1st Qu.:118.2   1st Qu.:6.300   1st Qu.: 57.84  
 Mode  :character   Median :215.4   Median :7.100   Median :125.89  
                    Mean   :220.3   Mean   :7.024   Mean   :141.93  
                    3rd Qu.:293.0   3rd Qu.:8.000   3rd Qu.:191.18  
                    Max.   :460.9   Max.   :8.800   Max.   :423.84  
      Meta            Len             Win             Nom       
 Min.   :25.00   Min.   : 89.0   Min.   : 1.00   Min.   : 1.00  
 1st Qu.:45.00   1st Qu.: 98.0   1st Qu.: 3.00   1st Qu.: 7.00  
 Median :64.00   Median :115.0   Median : 9.00   Median :15.00  
 Mean   :63.08   Mean   :114.7   Mean   :13.64   Mean   :19.56  
 3rd Qu.:81.00   3rd Qu.:126.0   3rd Qu.:18.00   3rd Qu.:28.00  
 Max.   :92.00   Max.   :157.0   Max.   :56.00   Max.   :74.00  

# 3.3 MOVIES - All Term Model

In [83]:
lmBoxMulti <- lm(Box~Rate + User + Meta + Len + Win + Nom , movies_df)
summary(lmBoxMulti) 


Call:
lm(formula = Box ~ Rate + User + Meta + Len + Win + Nom, data = movies_df)

Residuals:
    Min      1Q  Median      3Q     Max 
-53.161 -22.013  -3.864   9.517  84.574 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)  
(Intercept) -172.28110  106.51894  -1.617   0.1232  
Rate          35.34769   22.44744   1.575   0.1327  
User           0.38894    0.19304   2.015   0.0591 .
Meta           1.25615    0.89110   1.410   0.1757  
Len            0.02473    0.54429   0.045   0.9643  
Win           -0.02080    1.33384  -0.016   0.9877  
Nom            0.37261    0.87286   0.427   0.6745  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 42.45 on 18 degrees of freedom
Multiple R-squared:  0.8854,	Adjusted R-squared:  0.8472 
F-statistic: 23.18 on 6 and 18 DF,  p-value: 1.505e-07


## Nested Model Test
\begin{equation*}
\frac{RSS_R-RSS_C/(k-r)}{RSS_C/n-k-1}
\end{equation*}

Suppose that the full k-predictor model, also known as the complete model, has an RSS value equal to RSS$_{C}$.

Consider removing a subset of the predictor variables that we suspect provides little in- formation about the response, Y, beyond the information provided by the other predictors. Removing this subset leads to a reduced model with r predictors (i.e., k—r predictors are removed).

Since the reduced model is nested in the complete model (i.e., it contains a subset of the complete model predictors), it will have an RSS value, say, RSS$_{R}$, that is greater than or equal to RSS$_{C}$; see page 95 for a geometrical argument for why this is so.

Intuitively, if the difference between R S S R and RSSc is small, then the explanatory power of the two models is similar, and we would prefer the simpler reduced model since the complete model seems to be overfitting the sample data. On the other hand, if the difference between RSSR and RSSc is large, we would prefer the complete model since the k—r extra predictors in the complete model do appear to provide useful information about the response, Y, beyond the information provided by the rreduced model predictors.

To turn this intuition into a formal hypothesis test, we need to find a test statistic propor- tional to RSSR—RSSc, whose sampling distribution we know under a null hypothesis that states the reduced and complete models are equivalent in the population. The F-distribution we introduced in Section 3.3.3 serves this purpose in the following nested model test.

### First lets calculate the residual sum of squares for the complete model and the nested model.

In [152]:
rss_c <- deviance(lmBoxMulti)
cat("RSS for complete model: ", rss_c)

RSS for complete model:  32435.31

In [358]:
rss_r <- deviance(lmRateUserMeta)
cat("RSS for nested model: ", rss_r)

RSS for nested model:  32822.96

[1] 181.1711

In [347]:
# Nested Model Formula
numerator <- (rss_r - rss_c)/(6-3)
denominator <- rss_c/(25-6-1)
f_stat <- numerator/denominator

# Paramaters for Critical F
n_df <- 3 # same as r
d_df <- 18 # same as n-k-1
f_crit <- 3.160

In [348]:
cat("Calculated F-Statistic: ", f_stat,". Critical F Value: ", f_crit)

Calculated F-Statistic:  0.0717099 . Critical F Value:  3.16

### Therefore we fail to reject the Nested Model F-Test that: b4=b5=b6=0
## We can confirm our RSS values using the values in the ANOVA table below

In [357]:
anova(lmBoxMulti)
anova(lmRateUserMeta)

Df Sum Sq       Mean Sq      F value      Pr(>F)      
Rate       1 232048.12930 232048.12930 128.77529556 1.233877e-09
User       1  14477.47559  14477.47559   8.03428670 1.099249e-02
Meta       1   3748.57391   3748.57391   2.08027410 1.663884e-01
Len        1     38.03622     38.03622   0.02110823 8.860990e-01
Win        1     21.24289     21.24289   0.01178876 9.147400e-01
Nom        1    328.37631    328.37631   0.18223270 6.745238e-01
Residuals 18  32435.30764   1801.96154           NA           NA

Df Sum Sq     Mean Sq    F value    Pr(>F)      
Rate       1 232048.129 232048.129 148.463462 5.492367e-11
User       1  14477.476  14477.476   9.262631 6.175760e-03
Meta       1   3748.574   3748.574   2.398323 1.364049e-01
Residuals 21  32822.963   1562.998         NA           NA

### We can also use these values to calculate our Regression Standard Error (S)

In [262]:
# Formula for F-Statistic
MSR <- sum(anova(lmBoxMulti)$'Mean Sq')/6 # Variation between groups
MSE <- 1801.96154 # Variation within groups 
MSR/MSE

[1] 23.35083

### ANOVA and Regression are actually mathamatically very similar under the hood

In [344]:
# Handy ways to extract the residual sum of squares for anova.
cat("Deviance:",sum(lmBoxMulti$residuals^2),
    ". Also Deviance can be calculated using: ", deviance(lmBoxMulti), '\n')

print("Mean Sq is the Square Root of S:")
cat("Full Model S: ", sqrt(1801.96154), '\n')
cat("Partial Model S: ", sqrt(1562.998))

Deviance: 32435.31 . Also Deviance can be calculated using:  32435.31 
[1] "Mean Sq is the Square Root of S:"
Full Model S:  42.44952 
Partial Model S:  39.53477